### PR TITLE
Manage put client requests

### PR DESCRIPTION
**Pull request by netsirius**
Added logic to manage put client requests

### CODE REVIEW

File: crates/locutus-stdlib/src/interface.rs [hunk 1]
```diff
@@ -78,7 +78,26 @@ impl UpdateModification {
 
 #[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Default)]
 pub struct RelatedContracts {
-    map: HashMap<ContractInstanceId, Option<State<'static>>>,
+    pub map: HashMap<ContractInstanceId, Option<State<'static>>>,
```
Comment (iduartgomez) [inline crates/locutus-stdlib/src/interface.rs:81]: this should be at much `pub(crate)` , but not compeltely public sicne we want to hide impl details

File: crates/locutus-core/src/client_events.rs [hunk 2]
```diff
@@ -614,4 +647,37 @@ pub(crate) mod test {
         assert_eq!(result_client_request, expected_client_request);
         Ok(())
     }
+
+    #[test]
+    fn test_handle_put_request() -> Result<(), Box<dyn std::error::Error>> {
+        let contract = get_test_contract(TEST_CONTRACT_1);
```
Comment (iduartgomez) [inline crates/locutus-core/src/client_events.rs:653]: the test contract is not used or necessary here for anything

File: crates/locutus-core/src/client_events.rs [hunk 3]
```diff
@@ -281,8 +281,20 @@ impl ClientRequest {
                 let mut map_keys = Vec::from_iter(value_map.keys().copied());
                 map_keys.sort();
                 match map_keys.as_slice() {
-                    ["contract", "state"] => {
-                        todo!("Not implemented, this transformation needs to be implemented")
+                    ["contract", "relatedContracts", "state"] => {
+                        log::info!("Received put request");
```
Comment (iduartgomez) [inline crates/locutus-core/src/client_events.rs:285]: remove this log

File: crates/locutus-core/src/client_events.rs [hunk 4]
```diff
@@ -281,8 +281,20 @@ impl ClientRequest {
...
+                            state: WrappedState::try_from(*value_map.get("state").unwrap())
+                                .map_err(ErrorKind::deserialization)?,
+                            related_contracts: RelatedContracts::try_from(
+                                *value_map.get("relatedContracts").unwrap(),
+                            )
+                            .map_err(ErrorKind::deserialization)?,
+                        }
                     }
                     ["data", "key"] => {
                         log::info!("Received update request");
```
Comment (iduartgomez) [inline crates/locutus-core/src/client_events.rs:300]: remove this log

File: crates/locutus-stdlib/src/interface.rs (hunk 1 above)
Comment (netsirius) [reply crates/locutus-stdlib/src/interface.rs]: done

File: crates/locutus-core/src/client_events.rs (hunk 3 above)
Comment (netsirius) [reply crates/locutus-core/src/client_events.rs]: done

File: crates/locutus-core/src/client_events.rs (hunk 4 above)
Comment (netsirius) [reply crates/locutus-core/src/client_events.rs]: done

**Pull request merged**